### PR TITLE
Add temperature reading

### PR DESCRIPTION
**Pull request by zeeZ**
These are the bits I added locally to get a temperature reading:
image

Two points:
- I'm not very familiar with HA internals, so I don't know if the entity settings are correct.
- The controller would error due to the oldvalue key not existing until stream data arrives, so I made it an optional None with the `.get()` instead. I don't know of this how this affects the `lifetimeEnergy ` bit.

I do not know if this 100% works as is, especially on a fresh setup. Currently it's "works on my machine", but I might have time to actually test soonish.

### CODE REVIEW

**Comment by olalid** [comment]
Thanks for contributing!

I have not tried it yet, but looks like a good approach.
I am thinking we should add also temperature sensor for equalizer (state_internalTemperature).
There also seems to be more details available that we can add later as attributes to the charer temperature sensor, but that requires changes in the lib first so we can wait with that until we have something more pressing to change in the lib.
(e.g. TempInputT2, TempOutputN etc).

File: custom_components/easee/const.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
     ENERGY_WATT_HOUR,
     POWER_KILO_WATT,
     POWER_WATT,
+    TEMP_CELSIUS,
```
Comment (astrandb) [inline custom_components/easee/const.py:11]:
```suggestion
    UnitOfTemperature,
```

File: custom_components/easee/const.py [hunk 2]
```diff
@@ -60,6 +61,17 @@
     },
 }
 OPTIONAL_EASEE_ENTITIES = {
+    "temp_max": {
+        "key": "state.tempMax",
+        "attrs": [],
+        "units": TEMP_CELSIUS,
```
Comment (astrandb) [inline custom_components/easee/const.py:67]:
```suggestion
        "units": UnitOftemperature.CELSIUS,
```

**Review by astrandb — CHANGES_REQUESTED** [review]
Looks good to me.

File: custom_components/easee/const.py [hunk 3]
```diff
@@ -60,6 +61,17 @@
...
+        "convert_units_func": "round_0_dec",
+        "device_class": SensorDeviceClass.TEMPERATURE,
+        "state_class": SensorStateClass.MEASUREMENT,
+        "icon": "mdi:temperature-celsius",
```
Comment (astrandb) [inline custom_components/easee/const.py:71]:
```suggestion
```
This icon is wrong if user changes the unit of measurement. Remove and let HA use standard icon based on device_class.

**Review by astrandb — CHANGES_REQUESTED** [review]

**Comment by zeeZ** [comment]
I've added the units related changes (and totally didn't forget to lint).

On testing with a fresh empty instance, on startup it produces the following warning once per charger, and I'm not really sure what's happening there:

> WARNING (MainThread) [homeassistant.components.sensor] Sensor sensor.easee_home_xxxxxx_temp_max has device class temperature, state class measurement and unit °C thus indicating it has a numeric value; however, it has the non-numeric value:  (<class 'str'>); Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

I don't see where it could be getting a string from, I'm expecting an int from pyeasee or a None.

**Comment by olalid** [comment]
Perhaps it is somehow because no value is yet set when the sensor is first created/updated?

(It is most likely transferred as a double I would think, rather than int, but I guess that is what you meant).

**Comment by zeeZ** [comment]
Yeah, no value exists at first and it is only created once the stream comes in, but the `str` still puzzles me.

> (It is most likely transferred as a double I would think, rather than int, but I guess that is what you meant).

The [documentation](https://developer.easee.cloud/docs/observation-ids) says it's a double, but it seems to actually be an integer:

`[custom_components.easee.controller] Callback EHXXXXXX 4 150 state_tempMax 21`

**Comment by olalid** [comment]
We are going to move to a new way to retrieve the initial states of the chargers/equalizers soon, I think we can probably retrieve the initial state of this sensor too with that approach since we will be able to list exactly what data we we need.
So I would not spend too much effort on trying to solve that initial state problem right now.

**Comment by astrandb** [comment]
This temperature sensor is included in PR #291

**Pull request closed without merging**